### ISSUE TITLE
Bug: ActionMenu can exceed the height of the viewport

### ISSUE THREAD

**Issue by scurker** [dequelabs/cauldron#2068]
### Steps to reproduce

If there's a large number of items in the action list, it can expand beyond the size of the viewport leading to overflowing items inaccessible to both mouse and keyboard users:

<img width="308" height="1327" alt="Image" src="https://github.com/user-attachments/assets/2d2d7cbe-119a-442e-893b-5be94a1107cd" />

### Expected Behavior

We should have the default `max-height` of the element be `80vh` while still allowing for content authors to override.

### Version

v6.20.x
